barretenberg/cpp/src/barretenberg/protogalaxy/protogalaxy_prover_internal.hpp

```diff
@@ -68,33 +65,81 @@ template <class ProverInstances_> class ProtogalaxyProverInternal {
      * can go unused. By skipping the basis extension entirely we avoid this unneccessary work.
      *
      * Tests indicates that utilizing ShortUnivariates speeds up the `benchmark_client_ivc.sh` benchmark by 10%
-     * @note This only works if ProverInstances::NUM == 2. The whole protogalaxy class would require substantial
-     * revision to support more PKs so this should be adequate for now
+     * @note This only works for two instances.
      */
-    using ShortUnivariates = typename Flavor::template ProverUnivariates<ProverInstances::NUM>;
+    using ShortUnivariates = typename Flavor::template ProverUnivariates<NUM_INSTANCES>;
 
     using ExtendedUnivariates =
         typename Flavor::template ProverUnivariatesWithOptimisticSkipping<ExtendedUnivariate::LENGTH,
-                                                                          /* SKIP_COUNT= */ ProverInstances::NUM - 1>;
+                                                                          /*skip_count=*/SKIP_COUNT>;
 
     using ExtendedUnivariatesType =
         std::conditional_t<Flavor::USE_SHORT_MONOMIALS, ShortUnivariates, ExtendedUnivariates>;
 
-    using TupleOfTuplesOfUnivariates =
-        typename Flavor::template ProtogalaxyTupleOfTuplesOfUnivariates<ProverInstances::NUM>;
+    using TupleOfTuplesOfUnivariates = typename Flavor::template ProtogalaxyTupleOfTuplesOfUnivariates<NUM_INSTANCES>;
     using TupleOfTuplesOfUnivariatesNoOptimisticSkipping =
-        typename Flavor::template ProtogalaxyTupleOfTuplesOfUnivariatesNoOptimisticSkipping<ProverInstances::NUM>;
+        typename Flavor::template ProtogalaxyTupleOfTuplesOfUnivariatesNoOptimisticSkipping<NUM_INSTANCES>;
 
     using RelationEvaluations = decltype(create_tuple_of_arrays_of_values<typename Flavor::Relations>());
 
-    static constexpr size_t NUM_SUBRELATIONS = ProverInstances::NUM_SUBRELATIONS;
-
     ExecutionTraceUsageTracker trace_usage_tracker;
 
     ProtogalaxyProverInternal(ExecutionTraceUsageTracker trace_usage_tracker = ExecutionTraceUsageTracker{})
         : trace_usage_tracker(std::move(trace_usage_tracker))
     {}
 
+    /**
+     * @brief For a fixed row index and each polynomial, construct univariates from the corresponding value
+     * from each prover instance over the domain 0, .., EXTENDED_LENGTH-1.
+     *
+     * @example If the row index is 2, and there are 4 prover instances, visually we have
```

Comment:
I know you're just reproducing this from its original location but I don't find this big comment very helpful do you? (I actually don't even understand the notation..). Just saying something like "constructs univariates that interpolate the values of each instance across a given row" seems perfectly fine. It also seems confusing to maintain this 4-instance example when we've hard-coded 2 everywhere